doc/tutorial/fcm_fundamentals

```diff
@@ -0,0 +1,859 @@
+                        the repository can grow very very fast, example files of these files are word 
+                        documents.
+                        These can be saved but it is best to save them once they are complete and 
+                        finalised.
+                        We may wish to write documentation in a text editor or LaTeX to maintain a 
+                        version controlled document.
+                    </li>
+                </ul>
+            </div>
+            <div class="page-header">
+                <h3>Diagrams to explain ideas and concepts</h3>
+            </div>
+            <div class="well">
+                <pre>
+&lt;root&gt;
+    |
+    |-- &lt;project 1&gt;
+    |       |
+    |       |-- trunk
+    |       |-- branches
+    |
+    |-- &lt;project 2&gt;
+    |       |
+    |       |-- trunk
+    |       |-- branches
+    |
+    |-- ...
+
+</pre>
+                The structure of a repository.
+                <br/>
+                <br/>
+                <pre>
+                &lt;Trunk&gt;
+                   |
+    &lt;Branch3&gt;------|---------&lt;Branch2&gt;------&lt;Branch1&gt;
+       |           |#1          |              |
+       |           |            |#2            |
+       |#3         |            |              |
+       |           |#4          |              |
+       |           |            |              |#5
+       |           |            |              |#6
+       |           |            |              |
+       |           |#7          |              |
+       |#8         |            |              |
+       |           |            |#9            |
+       |-----------|#10         |              |
+                   |#11---------|              |
+                   |                           |#12
+                   |#13------------------------|
+                   |
+                   |#14
+
+</pre>
+                The concept of revision number. Every change that occurs results in an increase
+                in the revision number, shown here by the <samp>#</samp>.
+                <br/>
+                <br/>
+                <img src="merging.png" id="I1">
+                <br/>
+                Best working practice for merging.
+                <br/>
+                The problem with the second method is that in the Branch 1 all of the branch 
+                changes are mixed with the trunk changes.
+                Unlike the best practice method where they are still separate and therfore it is
+                easy to see what changes belong to the branch and what have been done on the
+                trunk.
+                <br/>
+                <br/>
+                <h4 id="D1"><code>fcm revert</code> v <code>fcm merge --reverse --revision [M:N]</code></h4>
+                <br/>
+                <pre>
+Local/workin copy---&gt;Made changes---&gt;fcm st---&gt;Issue---&gt;fcm revert &lt;problem file(s)&gt;
```

Comment:
Could we get a short pre-amble about the different situations in which you would apply these rather than going straight for the kill with the commands themselves. Kind of Problem: X Solution Y sort of thing?